code/modules/mob/living/simple_animal/bot/bot.dm

```diff
@@ -77,7 +77,10 @@
 	var/control_freq = BOT_FREQ		// bot control frequency
 	var/bot_filter 				// The radio filter the bot uses to identify itself on the network.
 	var/bot_type = 0 //The type of bot it is, for radio control.
-	var/data_hud_type = DATA_HUD_DIAGNOSTIC_BASIC //The type of data HUD the bot uses. Diagnostic by default.
+	/// The type of data HUD the bot uses. Diagnostic by default.
+	var/data_hud_type = DATA_HUD_DIAGNOSTIC_BASIC
+	/// The actual HUD visible to the bot if player-controlled.
+	var/datum/atom_hud/data_hud
```

Comment:
I'm weary about adding a hard ref here. Is this necessary? What if you just do:
```dm
/mob/living/simple_animal/bot/Login()
	...
	var/datum/atom_hud/data_hud = GLOB.huds[data_hud_type]
	data_hud.add_hud_to(src)
	
/mob/living/simple_animal/bot/Destroy()
	...
	var/datum/atom_hud/data_hud = GLOB.huds[data_hud_type]
	data_hud.remove_hud_from(src)
```

Reply:
This should be good, I doublechecked with AI bots, entering and aghosting them then deleting, and entering and deleting without aghost. Only GC failure I found is the known /datum/mind-related one.

Sorry you basically had to re-write this whole PR for me 😆